#! /usr/bin/env python

PACKAGE_NAME="laserscan_to_pointcloud"
NODE_NAME="laserscan_to_pointcloud_assembler"
FILE_NAME="LaserScanToPointcloudAssembler"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("target_frame", str_t, 0, "Frame of reference in which the scans are going to be transformed", "/map")
gen.add("laser_scan_topic", str_t, 0, "Topic with LaserScans to assemble", "laser_scan")
gen.add("min_range_cutoff_percentage_offset", double_t, 0, "Value that is going to be multiplied with the min_range reported with in the laser scan to remove points too far or too close", 1.05, 0, 2)
gen.add("max_range_cutoff_percentage_offset", double_t, 0, "Value that is going to be multiplied with the max_range reported with in the laser scan to remove points too far or too close", 0.95, 0, 2)
gen.add("number_of_scans_to_assemble_per_cloud", int_t, 0, "Number of LaserScan to assemble for each PointCloud2 published", 10, 1)
exit(gen.generate(PACKAGE_NAME, NODE_NAME, FILE_NAME))
